File: tests/python/tests/read.py
from tests.base import IOBenchBase

class IOBenchRead(IOBenchBase):
  def test_full(self, file_path):
    file = open(file_path, 'r') # only read
    self.contents = file.read()

  def test_stream(self, file_path):
    file = open(file_path, 'r') # only read
    for line in file:
        self.contents = line

  def test_type(self):
    return 'read'

File: tests/ruby/tests/read.rb
require_relative 'base.rb'

module IOBench
  class Read < Base
    private
    def test_full(file_path)
      IO.read(file_path)
    end

    def test_stream(file_path)
      IO.foreach(file_path) do |line|
        line
      end
    end

    def test_type
      'read'
    end
  end
end